test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1080,7 +1080,11 @@ private void assertShardIndexCounter() {
             IndicesService indexServices = getInstance(IndicesService.class, nodeAndClient.name);
             for (IndexService indexService : indexServices) {
                 for (IndexShard indexShard : indexService) {
-                    assertThat("index shard counter on shard " + indexShard.shardId() + " on node " + nodeAndClient.name + " not 0", indexShard.getActiveOperationsCount(), equalTo(0));
+                    try {
+                        assertBusy(() -> assertThat("index shard counter on shard " + indexShard.shardId() + " on node " + nodeAndClient.name + " not 0", indexShard.getActiveOperationsCount(), equalTo(0)));
```

Comment:
The assertion trips on my CI when I don't put assertBusy() there. I'll have to investigate why.